zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/BpmnProcessors.java

```diff
@@ -119,7 +121,11 @@ private static void addProcessInstanceCommandProcessor(
     typedRecordProcessors.onCommand(
         ValueType.PROCESS_INSTANCE,
         ProcessInstanceIntent.CANCEL,
-        new ProcessInstanceCancelProcessor(processingState, writers));
+        new AuthorizableProcessDefinitionProcessor<>(
+            new AuthorizationCheckBehavior(
+                processingState.getAuthorizationState(), processingState.getUserState()),
+            writers,
+            new ProcessInstanceCancelProcessor(processingState, writers)));
   }
```

Comment:
🔧 You could probably move the construction of `AuthorizationCheckBehavior` into the constructor of `AuthorizableProcessDefinitionProcessor` to clarify that the `ProcessInstanceCancelProcessor` is decorated.